src/hotspot/share/utilities/elfFile.cpp

```diff
@@ -1625,6 +1626,20 @@ bool DwarfFile::LineNumberProgram::get_filename_from_header(const uint32_t file_
   return false;
 }
 
+// Remove everything before the last slash including the slash itself to get the actual filename. This is required, for
+// example, for Clang debug builds which emit a relative path while GCC only emits the filename.
+void DwarfFile::LineNumberProgram::strip_path_prefix(char* filename, const size_t filename_len) {
+  char* last_slash = strrchr(filename, '/');
+  if (last_slash != nullptr) {
+    uint16_t index_after_slash = (uint16_t)(last_slash + 1 - filename);
+    // Copy filename to beginning of buffer.
+    int bytes_written = jio_snprintf(filename, filename_len - index_after_slash, "%s", filename + index_after_slash);
+    assert(bytes_written > 0, "could not strip path prefix");
+    // Add null terminator.
+    jio_snprintf(filename + bytes_written, 1, "%s", '\0');
```

Comment:
I think the correct format string is `%c` here. I also looked into other code that appends the `\0`, that one just pokes it in via direct assignment, but `snprintf` does not seem wrong.

Reply:
Right, directly setting it is more straight forward.